salalib/shapemap.cpp

```diff
@@ -3658,16 +3673,20 @@ void ShapeMap::ozlemSpecial2(ShapeMap& buildings)
       }
       if (!chopped && meencountered) {
          // section it here!
-         removePolyPixels(m_shapes.key(i));
-         m_shapes[i].m_region = Line(mepoint,li.t_end());
-         makePolyPixels(m_shapes.key(i));
+         removePolyPixels(shape.first);
+         m_shapes.find(i)->second.m_region = Line(mepoint,li.t_end());
+         makePolyPixels(shape.first);
       }
    }
 
    for (i = 0; i < removelist.size(); i++) {
-      removePolyPixels(m_shapes.key(removelist[i]));
+      removePolyPixels(depthmapX::getMapAtIndex(m_shapes, removelist[i])->first);
+   }
+   for (i = 0; i < removelist.size(); i++) {

```

Comment:
Not sure this works as intended - erasing tends to mess up the map iterators, and erasing from the front will make indices invalid.